Enable Auto-Formatting

rnovatorov opened the pull request:
Formatting conventions are effortlessly enforced using auto-formatting. It saves waste amounts of time and mental energy of both developers and reviewers.

This MR enables auto-formatting of Lua code using [StyLua](https://github.com/JohnnyMorganz/StyLua). The project seems to be actively maintained.

See also:

- https://github.com/Enapter/marketplace/pull/195#discussion_r1118420872

kulti reviewed (changes requested):
Too much changes and I'm not sure that all of them are good.

I think this linter should check only changed files. We can reformat and refactor blueprints one by one or by small batch. E.g. we need to rewrite some blueprint to use `enapter.ucm.config`.

Examples of not good or strange changes (in my opinion) is bellow.

It replace multiline comment (--[[...]]--) with multiple single line comments in one place, but in another change only end of multiline comment from:
```
]]--
```
to strange:
```
]]
--
```

Also look at boilers/ems-esp/firmware.lua. `scnum_list` changes are not looking good.

rnovatorov commented:
@kulti, thank you for your comment :)

I cannot think of a simple way to exclude files from being reformatted/checked. I suppose we would be better off spending time reviewing the changes than inventing a mechanism to format a subset of files.

> It replace multiline comment (--[[...]]--) with multiple single line comments in one place, but in another change only end of multiline comment from:
> 
> ]]--

This is my fault, I accidentally included manual changes in the commit that does auto-formatting. It was actually another linter complaining about the commented out code. I moved this changes into separate commit to be explicit.

> It replace multiline comment (--[[...]]--) with multiple single line comments in one place, but in another change only end of multiline comment from:
>
> ]]--
> to strange:
>
> ]]
> \--

Well, the [Lua syntax for multi-line comments](https://www.lua.org/pil/1.3.html) is `--]]` followed by a comment followed by `]]`. So this `--` in the end is a redundant single-line comment.

As for `scnum_list` changes, such long lists of values may be moved down the file to avoid distracting the reader, the same way as we move `main()` down the file.

kulti commented:
> @kulti, thank you for your comment :)
> 
> I cannot think of a simple way to exclude files from being reformatted/checked. I suppose we would be better off spending time reviewing the changes than inventing a mechanism to format a subset of files.

I agree with that. Pre-commit already works with changed files only and for CI I can suggest a solution in the following way:
```
      - name: Get Changed Files
        id: changes
        run: |
          echo "changed-files=$(git diff --name-only ${{ github.event.pull_request.head.sha }} ${{ github.sha }} | xargs)" >> "$GITHUB_OUTPUT"
      - name: Check Formatting
        uses: JohnnyMorganz/stylua-action@v2
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          version: v0.16.1
          args: --check --allow-hidden ${{ steps.changes.outputs.changed-files }}
```

rnovatorov commented:
@kulti, I have removed all the auto-formatting changes, adjusted the CI check to run only if some files have changed, updated README and created [an issue to track the review progress](https://github.com/Enapter/marketplace/issues/199). I think this MR can now be merged.

kulti reviewed (approved):
LGTM
